### ISSUE TITLE
Adding perplexity metric and long evals on last step

### ISSUE THREAD

**Issue by ruimalheiro** [ruimalheiro/gradient-garden#43]
We need to add the perplexity metric `(exp(global mean cross entropy loss))` which is easier to observe and also creates a more interpretable loss decay to monitor. Also since we already track last step evals, we could add the config for that specific last step, that we can run longer evals at that stage.

Expected work:
- Add perplexity metric to both train and val losses (both losses are already global mean and account for `ignore_index` so the calculation should be direct);
- Address logging modifications (console, file logging, WandB);
- Add config for longer evals in the last step. Each eval can be configured how many examples to run on last step: `number_of_examples_last_step`. If not set we can fall back to the `number_of_examples` property. (for validation it would be: `validation_steps_last_step` and fall back to `validation_steps`)
